schemas/iso19139/src/main/plugin/iso19139/schematron/schematron-rules-iso.sch

```diff
@@ -348,21 +348,19 @@ USA.
 			<sch:report test="gmd:aggregateDataSetName or gmd:aggregateDataSetIdentifier"
 				>$loc/strings/report.M23</sch:report>
 
-
-
-      <sch:let name="mdRefRef" value="gmd:aggregateDataSetIdentifier/*/gmd:code/*"/>
+      <sch:let name="mdRefRef" value="gmd:aggregateDataSetIdentifier/*/gmd:code/gco:CharacterString"/>
```

Comment:
I'm not sure if `gmx:Anchor` can be used instead of `gco:CharacterString`. If so, would be good to update the code to support both.

Similar comment about the change below.

Reply:
Going off the schemas contained in geonetwork and my current knowledge of the schemas, it seems `gmx:Anchor` is never an option here:

```xml
<gmd:aggregateDataSetIdentifier>
    <gmd:MD_Identifier>
        <gmd:code>
            <gco:CharacterString>9ff44cc4-5f16-4507-81a6-6810958b14df</gco:CharacterString>
        </gmd:code>
    </gmd:MD_Identifier>
</gmd:aggregateDataSetIdentifier>

<xs:complexType name="MD_Identifier_Type">
  <xs:complexContent>
    <xs:extension base="gco:AbstractObject_Type">
      <xs:sequence>
        <xs:element name="authority" type="gmd:CI_Citation_PropertyType" minOccurs="0"/>
        <xs:element name="code" type="gco:CharacterString_PropertyType"/>
      </xs:sequence>
    </xs:extension>
  </xs:complexContent>
</xs:complexType>

<xs:complexType name="CharacterString_PropertyType">
  <xs:sequence minOccurs="0">
    <xs:element ref="gco:CharacterString"/>
  </xs:sequence>
  <xs:attribute ref="gco:nilReason"/>
</xs:complexType>
```

Reply:
@joachimnielandt 

`gmx:Anchor` is a replacement of `gco:CharacterString`:

```
<xs:element name="Anchor" type="gmx:Anchor_Type" substitutionGroup="gco:CharacterString"/>
```

But it can be that GeoNetwork doesn't use it for the metadata created in the editor when adding the `gmd:aggregateDataSetIdentifier` element.

Reply:
Forgot about substitionGroups, thanks for reminding me! I have added `gmx:Anchor` as an option and added a unit test to check the various combinations.